### ISSUE TITLE
Provide an original size to images in FO

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#25790]
#### Is your feature request related to a problem?

I would like to allow showing full size product images in a fancybox on shop frontends. Possibly implement it also in a new PS default theme.

#### Describe the solution you'd like

- We currently have a bySize, small, medium and large sizes available for each presented image.
- I would like to have an original size available.
- URL looks like this, XXX is the image ID - https://www.domain.com/XXX/product-name-rewrite.jpg
- We should also provide the dimensions of the original image. We can get it by from the local image in like 0.0000009 seconds or we can save it into database.

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by MatShir** [comment]
What would be the interest on that? 
Even if you can access to the original format directly, what would be the interest if you can't use the ideal format, depending on browser and configuration?

**Comment by dennispw** [comment]
One benefit would be CDN services that have on-the-fly image generation. For a few clients we've taken the original image and generated different formats, ratios, greyscale versions, compression and much more. Just take a quick look at services such as IMGIX or KeyCDN, just to name two.

Since these services *usually* charges by the image, it's better to use the original images uploaded since they will only be charged once, instead of adding parameters for every single generate image class.

**Comment by Hlavtox** [comment]
@MatShir To have full screen product galleries like here:
https://eshop-franke.cz/trouby/100468-trouba-franke-mythos-fmy-98-p-bk-7612985785204.html

**Comment by MatShir** [comment]
Can't we create a new image setting with a big width and height? Use this size as original image 🤔

**Comment by Hlavtox** [comment]
@MatShir Yes I was thinking about it, it would be good. But there would need to be an option called "do not upscale image if the original is smaller" and "keep the aspect ratio".

**Comment by Hlavtox** [comment]
@MatShir @matthieu-rolland @NeOMakinG Closing this in favor of https://github.com/PrestaShop/PrestaShop/issues/26158.

This behavior I want would be achievable with "Keep original dimensions and respect the longest side of image". Plus, we can get a benefit of having it in proper (multiple) formats and ability to scale it down. The original image could be 10000x10000px, we don't want to load it in FO.

**Comment by dennispw** [comment]
> @MatShir @matthieu-rolland @NeOMakinG Closing this in favor of #26158.
> 
> This behavior I want would be achievable with "Keep original dimensions and respect the longest side of image". Plus, we can get a benefit of having it in proper (multiple) formats and ability to scale it down. The original image could be 10000x10000px, we don't want to load it in FO.

While I agree that it should never be loaded in FO, having the original image source and details *available* is still a good thing? And https://github.com/PrestaShop/PrestaShop/issues/26158 isn't *really* the same issue as this, since it's more about *how* to scale the image according to the available image classes?

I can't see any downsides to *having it available*. Right now if we need it (such as together with a CDN service like I mentioned earlier), we have to either resort to some Smarty shenanigans or create an override and/or module to fix it. I would much rather prefer something like `$image.bySize.original.src` - *if and/or when* needed.

**Comment by Hlavtox** [comment]
OK! Let's keep it available, we discussed with Mateus and it does not hurt. :-)

But for galleries like photoswipe, it will be better to use a thumbnail in original aspect ratio, with an upper size limit.

**Comment by Hlavtox** [comment]
Still relevant

**Comment by boo-code** [comment]
This thread holds two different requests, and one of them has now been delivered while the other has not - worth separating before deciding what is left.

@Hlavtox's gallery case is covered. #26158 was merged and is now closed as completed, and it added the fitment modes: `src/Core/Domain/ImageSettings/ValueObject/ImageFitment.php` defines `FIT`, `CROP` and `BOUND`, with `bound` being "keep original dimensions and respect the longest side" and not upscaling - which is what you said in 2022 would give you the photoswipe case, with the added benefit of the generated formats.

@dennispw's CDN case is not covered. The presented image array is still

```php
return [
    'bySize' => $urls,
    'small' => $small,
    'medium' => $medium,
    'large' => $large,
    'legend' => ...,
    'id_image' => $id_image,
];
```

and `small`, `medium` and `large` are only picks out of the configured image types - the first, middle and last of `bySize`. The uploaded file itself is read inside `ImageRetriever` at lines 169 and 205 purely to generate those thumbnails, and is never exposed. So a theme still cannot hand an on-the-fly service such as imgix or KeyCDN the source URL, which was the whole point of that half of the request.

PR #31034 attempted exactly this and was closed without merging, so nothing has replaced it.
